openwisp_utils/tests.py

```diff
@@ -11,3 +15,65 @@ def catch_signal(signal):
     signal.connect(handler)
     yield handler
     signal.disconnect(handler)
+
+
+class TimeLoggingTestResult(TextTestResult):
+    slow_test_threshold = getattr(settings, 'OPENWISP_SLOW_TEST_THRESHOLD', 0.3)
+
+    def __init__(self, *args, **kwargs):
+        self.test_timings = []
+        super().__init__(*args, **kwargs)
+
+    def startTest(self, test):
+        self._start_time = time()
+        super().startTest(test)
+
+    def addSuccess(self, test):
+        elapsed = time() - self._start_time
+        name = self.getDescription(test)
+        self.test_timings.append((name, elapsed))
+        super().addSuccess(test)
+
+    def color(self, color):
+        color_dict = {
+            'white_bold': '37;1',
+            'green_bold': '32;1',
+            'yellow_bold': '33;1',
+            'red_bold': '31;1',
+            'reset': '0',
+        }
+        assert color in color_dict
+        return f'\033[{color_dict[color]}m'
```

Comment:
Instead of asserting that the key exists in the dict, let's return a safe default like

```
color_dict = {...}
color = color_dict.get(color_name, '0')
return f'\033[{color}m'
```